indexing-service/src/test/java/io/druid/indexing/overlord/http/OverlordResourceTest.java

```diff
@@ -0,0 +1,305 @@
+        .retryPolicy(new RetryOneTime(1))
+        .compressionProvider(new PotentiallyGzippedCompressionProvider(true))
+        .build();
+  }
+
+  private void tearDownServerAndCurator()
+  {
+    CloseQuietly.close(curator);
+    CloseQuietly.close(server);
+  }
+
+  @Before
+  public void setUp() throws Exception
+  {
+    taskLockbox = EasyMock.createStrictMock(TaskLockbox.class);
+    taskLockbox.syncFromStorage();
+    EasyMock.expectLastCall().atLeastOnce();
+    taskLockbox.unlock(EasyMock.<Task>anyObject());
+    EasyMock.expectLastCall().atLeastOnce();
+    taskActionClientFactory = EasyMock.createStrictMock(TaskActionClientFactory.class);
+    EasyMock.expect(taskActionClientFactory.create(EasyMock.<Task>anyObject()))
+            .andReturn(null).anyTimes();
+    EasyMock.replay(taskLockbox, taskActionClientFactory);
+
+    taskStorage = new HeapMemoryTaskStorage(new TaskStorageConfig(null));
+    taskCountDownLatches = new CountDownLatch[2];
+    taskCountDownLatches[0] = new CountDownLatch(1);
+    taskCountDownLatches[1] = new CountDownLatch(1);
+    announcementLatch = new CountDownLatch(1);
+    IndexerZkConfig indexerZkConfig = new IndexerZkConfig(new ZkPathsConfig(), null, null, null, null, null);
+    setupServerAndCurator();
+    curator.start();
+    curator.create().creatingParentsIfNeeded().forPath(indexerZkConfig.getLeaderLatchPath());
+    druidNode = new DruidNode("hey", "what", 1234);
+    ServiceEmitter serviceEmitter = new ServiceEmitter(null, null, null){
+      @Override
+      public void emit(Event event) {
+        // NOOP
+      }
+    };
+    taskMaster = new TaskMaster(
+        new TaskQueueConfig(null, new Period(1), null, new Period(100)),
+        taskLockbox,
+        taskStorage,
+        taskActionClientFactory,
+        druidNode,
+        indexerZkConfig,
+        new TaskRunnerFactory()
+        {
+          @Override
+          public TaskRunner build()
+          {
+            return new MockTaskRunner(taskCountDownLatches);
+          }
+        },
+        new ResourceManagementSchedulerFactory()
+        {
+          @Override
+          public ResourceManagementScheduler build(
+              RemoteTaskRunner runner, ScheduledExecutorFactory executorFactory
+          )
+          {
+            return new NoopResourceManagementScheduler();
+          }
+        },
+        curator,
+        new NoopServiceAnnouncer(){
+          @Override
+          public void announce(DruidNode node)
+          {
+            announcementLatch.countDown();
+          }
+        },
+        serviceEmitter
+    );
+    EmittingLogger.registerEmitter(serviceEmitter);
+  }
+
+  @Test
+  public void testOverlordResource() throws Exception{
+    // basic task master lifecycle test
+    taskMaster.start();
+    Assert.assertTrue(announcementLatch.await(2, TimeUnit.SECONDS));
+    // To be on safer side wait for "leading" boolean variable to be set to true
+    Thread.sleep(10);
```

Comment:
Can wait for the condition to be satisfied in a while loop and put a test timeout for this test method.